bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/protocol/UDPListenDiscoverRunnable.java

```diff
@@ -0,0 +1,124 @@
+ * @author Alessandro Del Pex - Souliss App
+ */
+@NonNullByDefault
+public class UDPListenDiscoverRunnable implements Runnable {
+
+    protected boolean bExit = false;
+    @Nullable
+    UDPDecoder decoder = null;
+
+    private final Logger logger = LoggerFactory.getLogger(UDPListenDiscoverRunnable.class);
+
+    @Nullable
+    private SoulissGatewayHandler gwHandler;
+
+    public UDPListenDiscoverRunnable(Bridge bridge, @Nullable DiscoverResult pDiscoverResult) {
+        this.gwHandler = (SoulissGatewayHandler) bridge.getHandler();
+        decoder = new UDPDecoder(bridge, pDiscoverResult);
+    }
+
+    @Override
+    public void run() {
+        DatagramSocket socket = null;
+
+        while (!Thread.currentThread().isInterrupted()) {
+            try {
+                // open socket for listening...
+                var channel = DatagramChannel.open();
+                socket = channel.socket();
+
+                socket.setReuseAddress(true);
+                socket.setBroadcast(true);
+
+                var localGwHandler = this.gwHandler;
+                if (localGwHandler != null) {
+                    var sa = new InetSocketAddress(localGwHandler.gwConfig.preferredLocalPortNumber);
+                    socket.bind(sa);
+
+                    var buf = new byte[200];
+                    // receive request
+                    final var packet = new DatagramPacket(buf, buf.length);
+                    socket.setSoTimeout(60000);
+                    socket.receive(packet);
+                    buf = packet.getData();
+
+                    // **************** DECODER ********************
+                    logger.debug("Packet received (port {}) {}", socket.getLocalPort(), macacoToString(buf));
+
+                    var localDecoder = this.decoder;
+                    if (localDecoder != null) {
+                        localDecoder.decodeVNetDatagram(packet);
+                    }
+                }
+
+            } catch (BindException e) {
+                logger.error("UDP Port busy, Souliss already listening? {} ", e.getLocalizedMessage());
+                try {
+                    if (socket != null && !socket.isClosed()) {
+                        socket.close();
+                    }
+                } catch (Exception e1) {
+                    logger.error("UDP socket close failed: {} ", e1.getLocalizedMessage());
+                }
+            } catch (SocketTimeoutException e2) {
+                logger.warn("UDP SocketTimeoutException close: {}", e2.getLocalizedMessage());
+                if (socket != null && !socket.isClosed()) {
+                    socket.close();
+                }
+            } catch (Exception ee) {
+                logger.error("Exception receiving-decoding message: {} ", ee.getLocalizedMessage());
+                if (socket != null && !socket.isClosed()) {
+                    socket.close();
+                }
+            } finally {
```

Comment:
If there's a network error, the thing status should be set to OFFLINE.

Reply:
cannot access updateStatus method from here. Any suggestion ? I must check socket close on the bridge ?

Reply:
As you have access to the gateway handler, you can add a method to the gateway handler which handles these exceptions. Or, you can simply send the error message as a string via an argument to the newly added method.

Reply:
added method to gatewayHandler to notify and put bridge offline.